chore(ci): move Go tests to parallel stage

MiguelPires opened the pull request:
## Description

Moves the Go tests into the parallel "Test" stage.

## Related issues

closes #3511 

## Pull Request Checklist

- [ ] All commit messages match our [commit message guidelines](https://github.com/zeebe-io/zeebe/blob/develop/CONTRIBUTING.md#commit-message-guidelines)
- [ ] The submitting code follows our [code style](https://github.com/zeebe-io/zeebe/wiki/Code-Style)
- [ ] If submitting code, please run `mvn clean install -DskipTests` locally before committing

MiguelPires commented:
Hey @Zelldon, thanks for the review. I moved the go build step into the parallel stage with the go test and left the other steps as a "prepare tests" stage since they're necessary for the tests to run

menski commented:
Please remember that we need the go build before the distro is build as it should contain the `zbctl` binaries.

MiguelPires commented:
True but I thought it wouldn't matter since this tag is only used for testing

Zelldon commented:
Ah I thought we build the distro and use it for the Zbctl tests? We could then copy the zbctl after the tests?

MiguelPires commented:
Currently, the Go build is being done after the Java build which I think is fine because they're done before the upload stage where `zbctl` is copied into the distro. So I think that's fine, right @Zelldon? However, I noticed that the SNAPSHOT docker image only has the linux `zbctl` binary, not the others. I'm not sure why this happens

MiguelPires commented:
The reason why the SNAPSHOT image only had the linux `zbctl` binary is because the binaries were being overwritten in the go tests. I changed the test to build all binaries and not just the linux one. I fixed this in the latest commit

Zelldon commented:
And the upload stage can access still the zbctl even when it is build in the parallel stage? If so then it is fine from my side. @menski any thoughts?

MiguelPires commented:
Yes, I think so but when the PR is merged I'll look into SNAPSHOT to be sure the zbctl binaries are copied correctly

MiguelPires commented:
bors r+

MiguelPires commented:
bors retry
